File: string/java/NaiveMatcher.java

public class NaiveMatcher implements StringMatcher
{
	public int match(String text, String pattern)
	{
		int j;
		for(int i = 0; i < text.length() - pattern.length(); i++)
		{
			for(j = 0; j < pattern.length(); j++)
				if(text.charAt(i + j) != pattern.charAt(j))
					break;
			if(j == pattern.length())
				return i;
		}
		return -1;
	}
}

File: string/py/naive_matcher.py
# Time:  O(nk)
# Space: O(1)

def naive_matcher(pattern, text):
    for i in range(len(text) - len(pattern) + 1):
        for j in range(len(pattern)):
            if text[i + j] != pattern[j]:
                break
        else:
            return i
    return -1
